lucet-module/src/error.rs

```diff
@@ -1,16 +1,16 @@
-use failure::Fail;
+use thiserror::Error;
 
 /// Module data (de)serialization errors.
-#[derive(Debug, Fail)]
+#[derive(Debug, Error)]
 pub enum Error {
-    #[fail(display = "Sparse data contained a page with length other than 4096")]
+    #[error("Sparse data contained a page with length other than 4096")]
     IncorrectPageSize,
-    #[fail(display = "Deserialization error: {}", _0)]
-    DeserializationError(#[cause] bincode::Error),
-    #[fail(display = "Serialization error: {}", _0)]
-    SerializationError(#[cause] bincode::Error),
-    #[fail(display = "Module signature error: {}", _0)]
-    ModuleSignatureError(#[cause] minisign::PError),
-    #[fail(display = "I/O error: {}", _0)]
-    IOError(#[cause] std::io::Error),
+    #[error("Deserialization error: {}", _0)]
+    DeserializationError(#[source] bincode::Error),
```

Comment:
Take a look at https://users.rust-lang.org/t/do-people-not-care-about-printable-error-chains-a-k-a-how-to-nicely-implement-display-for-an-error/35362 -- normally the inner error should not be included in both the Display impl of the outer error *and* the source, only one or the other, or else it will appear twice when printing out the full chain of causes.

The cause chain here currently would be:

```console
Error: Whatever top level thing failed

Caused by:
    - Deserialization error: whatever Bincode said failed
    - whatever Bincode said failed
```

Reply:
Thanks for citing this immensely helpful article!  I can take steps to avoid the annoying double-display.